addons/test/misra/misra-test.c

```diff
@@ -480,6 +480,9 @@ void misra_9_struct_initializers(void) {
     dummy_struct dse[3]    = { {1,2}, {3,4}, [1] = {5,6} };   // 9.3 9.4
     dummy_struct dsd[]     = { [0] = 1 };                     // 9.5
 
+    // Syntax error: initialization of an unnamed anonymous struct should not crash
+    struct { int j; } = { .j = 1 };
```

Comment:
I try to understand what AST we should create. how do I compile such code? Command `gcc -fsyntax-only 3176.c`:

```
3176.c:1:19: error: expected identifier or ‘(’ before ‘=’ token
 struct { int j; } = { .j = 1 };
                   ^
```

should we improve our syntax check and bailout?

Reply:
Hmm, I tried some more of my past examples, but I can't remember if there are any cases with the correct syntax that can reproduce crashes in `cppcheckdata.py`. Probably it fails only on incorrect code, because we trying to access the variable from `IdMap` which does not exists like in this code snippet in my test.

The only thing that confuses me it that cppcheck doesn't generate the warnings on syntax error. But probably it should not.
I think the similar cases will hardly ever occur in the real code base, which, in addition, tries to be MISRA-compliant

May be we should merge this PR "as is"? The user won't get the warnings about the incorrect syntax (the missing variable name after declaring an anonymous structure), but the addon will not crash.

Reply:
Thanks! I believe a syntax error from Cppcheck would be better in the long run. I can look at that solution soon..